File: Q30 Find the index of first occurance in a string.js
var strStr = function(haystack, needle) {
    let n = haystack.length;
    if (n === 0 && needle === "") {
        return 0;
    }

    let m = needle.length;
    for (let i = 0; i <= n - m; i++) {
        for (let j = 0; j < m; j++) {
            if (haystack[i + j] !== needle[j]) {
                break;
            }
            if (j === m - 1) {
                return i;
            }
        }
    }
    return -1;
};

File: Q30 find the index of first occurance in a string.py
class Solution:
    def strStr(self, haystack, needle) :
        n = len(haystack)
        if n == 0 and not needle:
            return 0

        m = len(needle)
        for i in range(n - m + 1):
            for j in range(m):
                if haystack[i + j] != needle[j]:
                    break
                if j == m - 1:
                    return i
        return -1

